tests/test_gui.py

```diff
@@ -14,3 +16,41 @@ def test_gui_home(client: TestClient):
 
     assert response.status_code == 200
     assert "Select Directory" in response.text
+
+
+def test_gui_navigate_success(
+    client: TestClient,
+    tmp_path: Path,
+):
+    response = client.get("/?path=%s" % str(tmp_path))
```

Comment:
See https://www.python-httpx.org/quickstart/#passing-parameters-in-urls . We can make the test client encode the query parameters for us, which is more readable and ensures that URL encoding happens properly.

Reply:
updated in cbdc6e68684f30990e1b2f5e8764d59ac3fbbebf